spec/expectations_hook_spec.rb

```diff
@@ -1,238 +1,323 @@
+        {expectation: expectations[2], result: false}] }
+  end
 
-    it { expect('program {}').not_to comply_with has_while_expectation }
+  describe 'Declares' do
+    let(:code) { "function foo(x, y) { return(x) }\nprocedure Bar(x) { }" }
+    let(:expectations) { [
+      {binding: '', inspection: 'Declares:foo'},
+      {binding: '', inspection: 'Declares:Bar'},
+      {binding: '', inspection: 'Declares:baz'}] }
+
+    it { expect(result).to eq [
+        {expectation: expectations[0], result: true},
+        {expectation: expectations[1], result: true},
+        {expectation: expectations[2], result: false}] }
+  end
 
-    let(:program_with_while) { '
+  context 'HasWhile expectation' do
+    let(:code) { %q{
       program {
         i := 3
         while(i > 0) {
           Mover(Oeste)
           i := i - 1
         }
-      }
-    ' }
+      } } }
 
-    it { expect(program_with_while).to comply_with has_while_expectation }
-  end
+    let(:expectations) { [{binding: 'program', inspection: 'HasWhile'}] }
 
-  context 'HasRepeatOf:n expectation' do
-    let(:has_repeat_of_1_expectation) { {'binding' => 'program', 'inspection' => 'HasRepeatOf:1'} }
+    it { expect(result).to eq [{expectation: {binding: 'program', inspection: 'UsesWhile'}, result: true}] }
+  end
 
-    let(:program_with_repeat_1) { '
+  context 'HasVariable expectation' do
+    let(:code) { %q{
       program {
-        repeat(1) {
-          Mover(Oeste)
-        }
-      }
-    ' }
+        acum := 25
+    }} }
 
-    let(:program_with_repeat_8) { '
-      program {
-        repeat(8) {
-          Mover(Oeste)
-        }
-      }
-    ' }
+    let(:expectations) { [
+      {'binding' => 'program', 'inspection' => 'DeclaresVariable'},
+      {'binding' => 'foo', 'inspection' => 'DeclaresVariable'}] }
 
-    it { expect(program_with_repeat_1).to comply_with has_repeat_of_1_expectation }
-    it { expect(program_with_repeat_8).not_to comply_with has_repeat_of_1_expectation }
-    it { expect('program { Mover(Oeste) }').not_to comply_with has_repeat_of_1_expectation }
+    it { expect(result).to eq [
+          {expectation: {inspection: 'DeclaresVariable', binding: 'program'}, result: true},
+          {expectation: {inspection: 'DeclaresVariable', binding: 'foo'}, result: false}] }
   end
 
-  context 'HasRepeat expectation' do
-    let(:has_repeat_expectation) { {'binding' => 'program', 'inspection' => 'HasRepeat'} }
+  context 'HasForeach expectation' do
+    let(:expectations) { [
+      {'binding' => 'program', 'inspection' => 'HasForeach'},
+      {'binding' => 'foo', 'inspection' => 'HasForeach'}] }
 
-    let(:program_with_repeat) { '
+    let(:code) { %q{
       program {
-        repeat(6) {
-          Mover(Oeste)
+        foreach color in [Azul..Verde] {
+          Poner(color)
         }
       }
-    ' }
+      function foo(){
+        return (1)
+      }}}
+
+    it {
+      pending "foreach is not yet supported by gobstones web"
```

Comment:
This is wrong: `foreach` is indeed supported, but ranges like `[Azul..Verde]` are not. You can test with `foreach color in [Azul, Negro, Rojo, Verde]` if you want.